src/brevitas/graph/gpfq.py

```diff
@@ -179,6 +180,9 @@ def single_layer_update(self):
 
         if hasattr(self.layer, 'offload_params'):
             self.layer.offload_params(self.layer)
+        # offload original weights onto the CPU
+        if self.create_weight_orig:
```

Comment:
Can you check if this is necessary? 
After you move `weight_orig` to device, can you check if `self.layer.weight_orig.device == weight_orig.device`?